### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by mallowmew**
* Adds another stat
* Refactors page layout function(s) to allow for better unit test coverage

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            $pageString .= 'default.svg">';
+        }
+        $pageString .= '<h1>' . $record['title'] . '</h1>';
+        if ( $record['finished'] >= 1) {
+            $pageString .= '<div class="finished-badge"></div>';
+        }
+        $pageString .= '</div>';
+        $year = $record['year-played'];
+    }
+    return $pageString;
```
Comment (MotNosniktaw) [inline functions.php]: this results in free floating non-contained divs for each thumbnail. is this a valid/appropriate html layout?

File: tests/testFunctions.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        $result = buildPage($mockData);
+    }
+
+    function testBuildPage_inputMalformed_singleRecordIndex() {
+        $mockData = [
+            'Another individual game', 
+            'isolated-boxart.jpg', 
+            2001, 
+            1
+        ];
```
Comment (MotNosniktaw) [inline tests/testFunctions.php]: is this redundant considering the test above? they both test for arrays that are not compatible with function

Comment (mallowmew) [reply tests/testFunctions.php]: I am not sure - they test different failures event hough the failure conditions return the same error. I will ask Sam.

File: tests/testFunctions.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
+<?php require '../functions.php';
+
+use PHPUnit\Framework\TestCase;
+
+class FunctionsTest extends TestCase {
```
Comment (sjr-mayden) [inline tests/testFunctions.php]: File name should match the name of the class that it contains, rename to FunctionsTest.php

File: tests/testFunctions.php [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    function testBuildPage_Success() {
```
Comment (sjr-mayden) [inline tests/testFunctions.php]: Methods in OOP need an access level

File: tests/testFunctions.php (hunk 2 above)
Comment (sjr-mayden) [reply tests/testFunctions.php]: Would say these are both valid tests. If it was the same code producing the same error then there would probably be further discussion but this is fine

File: functions.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+/**
+ * Builds the page.
+ * Takes the output array of getAllThumbnails. Builds a string laying out 
+ * the main page, separating groups of thumbnails by the year played.
+ *
+ * @param array $allRecords
+ * @return string
+ */
```
Comment (sjr-mayden) [inline functions.php]: Exceptions thrown by a function or method should be mentioned in the docbloc

File: functions.php [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    
+    $pageString = '';
+    $year = '';
+    foreach ($allRecords as $index => $record) {
+        if ($year !== $record['year-played']) {
+            $pageString .= '<div class="separator">'
+                . '<h1>' . ($record['year-played'] == null ? 'Unknown' : $record['year-played']) . '</h1>'
+                . '</div>';
+        }
+        $pageString .= '<div class="thumbnail"><img src = "./images/thumbnails/';
```
Comment (sjr-mayden) [inline functions.php]: img tag is missing an alt attribute

File: index.php [hunk 7]
```diff
@@ -1,85 +1,9 @@
...
-    $query = $db->prepare(
-        'SELECT `title`,`box-art`,`year-played` FROM `games-i-have-played`
-        ORDER BY `year-played` DESC');
-    $query->execute();
-    return $query->fetchAll();
-}
+require_once('functions.php');
 
+$db = connectToDB();
 $allThumbnails = getAllThumbnails($db);
```
Comment (sjr-mayden) [inline index.php]: Are they thumbnails or would games be a better description of what these are?

File: functions.php [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+/**
+ * Fetches stats for every record in the database.
+ * Returns a list with the most recently played titles at the top.
+ *
+ * @param PDO $db
+ * @return array
+ */
+function getAllThumbnails(&$db):array {
```
Comment (sjr-mayden) [inline functions.php]: Does the PDO object need to be passed by reference here?

**Pull request merged**